client_it_test.go

```diff
@@ -1502,7 +1502,15 @@ func TestUserAgent(t *testing.T) {
 }
 
 func TestClientRace(t *testing.T) {
-	client := getClient()
+	// Using a custom client with SkipVersionCheck to test race conditions in version checking
+	uri, err := NewURIFromAddress(getPilosaBindAddress())
+	if err != nil {
+		panic(err)
```

Comment:
why panic instead of `t.Fatal(err)` (or similar)?

Reply:
That condition should never fail unless the server address configuration for tests (or `NewURIFromAddress` function) is wrong so checking that error condition is equivalent to no-op in my mind. It also has the benefit of stopping the test run immediately if the server address config is wrong (which would make other tests fail anyway). Admittedly I wasn't consistent with using `t.Fatal`/panics/not checking errors` for `NewURIFromAddress` in the integration tests.

Reply:
I see - I'm not worried about it now, but `github.com/pilosa/pilosa/test` now has a function for setting up a pilosa cluster in-process, so at some point we should migrate the client integration tests to that rather than having `getPilosaBindAddress()` at all.

Reply:
That's cool. But wouldn't that make the client tests depend on the Pilosa server?

Reply:
yes it will, is that a problem? having just the tests depend on pilosa won't pull it into the client when used as a library.